lib/ansible/modules/cloud/cloudstack/cs_image_store.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2019, Patryk Cichy @PatTheSilent
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['stableinterface'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: cs_image_store
+
+short_description: Manages CloudStack Image Stores.
+
+version_added: "2.8"
+
+description:
+    - ""
+
+options:
+    url:
+        description:
+            - The URL for the Image Store.
+        required: true
+    name:
+        description:
+            - The ID of the Image Store. Required when deleting a Image Store.
+        required: true
+    zone:
+        description:
+            - The Zone name for the Image Store.
+        required: true
+    state:
+        description:
+            - Stage of the Image Store
+        choices: ['present', 'absent']
+        default: present
+    provider:
+        description:
+            - The image store provider name. Required when creating a new Image Store
+
+extends_documentation_fragment: cloudstack
+
+author:
+    - Patryk Cichy (@PatTheSilent)
+'''
+
+EXAMPLES = '''
+- name: Add a Image Store (NFS)
+  cs_image_store:
+    zone: zone-01
+    name: nfs-01
+    url: nfs://192.168.21.16/exports/secondary
+'''
+
+RETURN = '''
+id:
+    description: the ID of the image store
+    type: str
+    returned: success
+name:
+    description: the name of the image store
+    type: str
+    returned: success
+protocol:
+    description: the protocol of the image store
+    type: str
+    returned: success
+provider_name:
+    description: the provider name of the image store
+    type: str
+    returned: success
+scope:
+    description: the scope of the image store
+    type: str
+    returned: success
+url:
+    description: the url of the image store
+    type: str
+    returned: success
+zone:
+    description: the Zone name of the image store
+    type: str
+    returned: success
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import AnsibleCloudStack, cs_argument_spec, cs_required_together
+
+
+class AnsibleCloudstackImageStore(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(AnsibleCloudstackImageStore, self).__init__(module)
+        self.returns = {
+            'id': 'id',
+            'name': 'name',
+            'protocol': 'protocol',
+            'providername': 'provider_name',
+            'scope': 'scope',
+            'url': 'url',
+            'zone': 'zone'
```

Comment:
`id`, `name` and `zone` (which should be 'zonename': 'zone') are returned by default. Can be omitted in modules,

Reply:
My bad :)